glean-core/rlb/src/net/mod.rs

```diff
@@ -99,19 +100,25 @@ impl UploadManager {
                                 request.headers.into_iter().collect();
                             let result = inner.uploader.upload(upload_url, request.body, headers);
                             // Process the upload response.
-                            glean_core::glean_process_ping_upload_response(doc_id, result);
+                            match glean_core::glean_process_ping_upload_response(doc_id, result) {
+                                UploadTaskAction::Continue => continue,
+                                UploadTaskAction::End => break,
+                            }
                         }
                         PingUploadTask::Wait { time } => {
+                            log::trace!("Instructed to wait for {:?}ms", time);
```

Comment:
If we're tracing the other `PingUploadTask::*` we should trace `PingUploadTask::Upload`, no?